eclair-core/src/main/scala/fr/acinq/eclair/blockchain/WatcherTypes.scala

```diff
@@ -72,11 +72,18 @@ object WatchConfirmed {
  * @param outputIndex     index of the outpoint to watch.
  * @param publicKeyScript electrum requires us to specify a public key script; the script of the outpoint must be provided.
  * @param event           channel event related to the outpoint.
+ * @param hints           txids of potential spending transactions; most of the time we know the txs, and it allows for optimizations.
+ *                        This argument can safely be ignored by watcher implementations.
  */
-final case class WatchSpent(replyTo: ActorRef, txId: ByteVector32, outputIndex: Int, publicKeyScript: ByteVector, event: BitcoinEvent) extends Watch
+final case class WatchSpent(replyTo: ActorRef, txId: ByteVector32, outputIndex: Int, publicKeyScript: ByteVector, event: BitcoinEvent, hints: Seq[ByteVector32]) extends Watch
 object WatchSpent {
   // if we have the entire transaction, we can get the publicKeyScript from the relevant output
-  def apply(replyTo: ActorRef, tx: Transaction, outputIndex: Int, event: BitcoinEvent): WatchSpent = WatchSpent(replyTo, tx.txid, outputIndex, tx.txOut(outputIndex).publicKeyScript, event)
+  def apply(replyTo: ActorRef, tx: Transaction, outputIndex: Int, event: BitcoinEvent, hints: Seq[ByteVector32]): WatchSpent = WatchSpent(replyTo, tx.txid, outputIndex, tx.txOut(outputIndex).publicKeyScript, event, hints)
+
+  def watchFundingTx(channel: ActorRef, commitments: Commitments): WatchSpent = {
```

Comment:
I'd prefer putting that function in a channel-related file (why not in `Channel.scala` itself?).
Watcher/blockchain components don't really need to know about `commitments` and other channel-related data.

Reply:
Done with 3f69778.